Add serial terminal section to cli-collaborate.md

AnotherButler opened the pull request:
Add section from legacy documentation.

Commit:
Add serial terminal section to cli-collaborate.md

Add section from legacy documentation.

theotherjimmy commented:
PR #1000 :tada:

theotherjimmy commented on docs/tools/CLI/cli-collaborate.md:
```diff
@@ -101,6 +101,35 @@ $ mbed export -i uvision -m K64F
 
+#### Serial terminal
+
+You can open a serial terminal to the COM port of a connected Mbed target (usually board) using the `mbed sterm` command. If no COM port is specified, Mbed CLI will attempt to detect the connected Mbed targets and their COM ports.
```
`COM` port is a windows-only term. They're called `tty...` on linux and OSX. Could we replace COM with serial?

theotherjimmy commented on docs/tools/CLI/cli-collaborate.md:
```diff
@@ -101,6 +101,35 @@ $ mbed export -i uvision -m K64F
+
+There are various options to `mbed sterm`:
+
+- `--port <COM port>` to specify system COM port to connect to.
```
COM -> serial. again, to remove a windows-only term

theotherjimmy commented on docs/tools/CLI/cli-collaborate.md:
```diff
@@ -101,6 +101,35 @@ $ mbed export -i uvision -m K64F
+
+More shortcuts can be viewed within the serial terminal's help menu (Ctrl+h).
+
+You can also add the `--sterm` option to `mbed compile -f` to compile a new program, flash the program/firmware image to the connected target and then open the serial terminal to its COM port:
```
COM -> serial

Commit:
Replace references to COM in cli-collaborate.md

Replaces references to COM with serial, so the instructions don't relate only to Windows, as requested in the comments.
